pkg/logql/evaluator.go

```diff
@@ -302,21 +330,11 @@ func (ev *defaultEvaluator) vectorAggEvaluator(ctx context.Context, expr *vector
 	}, nextEvaluator.Close)
 }
 
-func (ev *defaultEvaluator) rangeAggEvaluator(ctx context.Context, expr *rangeAggregationExpr, q Params) (StepEvaluator, error) {
-	entryIter, err := ev.querier.Select(ctx, SelectParams{
-		&logproto.QueryRequest{
-			Start:     q.Start().Add(-expr.left.interval),
-			End:       q.End(),
-			Limit:     0,
-			Direction: logproto.FORWARD,
-			Selector:  expr.Selector().String(),
-		},
-	})
-
-	if err != nil {
-		return nil, err
-	}
-
+func rangeAggEvaluator(
```

Comment:
I love those changes.

Reply:
I wonder if evaluator is big enough to become a package. Just a thought, not action required.

Reply:
yeah, it's a good idea, let's see where everything settles and then do some refactoring